pkg/executor/sortexec/sort.go

```diff
@@ -97,30 +138,59 @@ func (e *SortExec) Open(ctx context.Context) error {
 		e.diskTracker = memory.NewTracker(e.ID(), -1)
 		e.diskTracker.AttachTo(e.Ctx().GetSessionVars().StmtCtx.DiskTracker)
 	}
+
+	e.IsUnparallel = !e.Ctx().GetSessionVars().EnableParallelSort
+	if e.IsUnparallel {
+		e.Unparallel.Idx = 0
+		e.Unparallel.PartitionList = e.Unparallel.PartitionList[:0]
+	} else {
+		e.Parallel.idx = 0
+		e.Parallel.workers = make([]*parallelSortWorker, e.Ctx().GetSessionVars().ExecutorConcurrency)
+		e.Parallel.mpmcQueue = chunk.NewMPMCQueue(chunk.DefaultMPMCQueueLimitNum)
+		e.Parallel.err = nil
+	}
+
 	e.sortPartitions = e.sortPartitions[:0]
-	return e.Children(0).Open(ctx)
+	return exec.Open(ctx, e.Children(0))
 }
 
 // Next implements the Executor Next interface.
-// Sort constructs the result following these step:
+// Sort constructs the result following these step in unparallel mode:
 //  1. Read as mush as rows into memory.
 //  2. If memory quota is triggered, sort these rows in memory and put them into disk as partition 1, then reset
 //     the memory quota trigger and return to step 1
 //  3. If memory quota is not triggered and child is consumed, sort these rows in memory as partition N.
 //  4. Merge sort if the count of partitions is larger than 1. If there is only one partition in step 4, it works
 //     just like in-memory sort before.
+//
+// TODO add introduction of parallel mode
```

Comment:
it would be better to add the introduction in this PR like https://github.com/pingcap/tidb/blob/4ed493dcc1e3d55dd5af46db49d0b6cd34a24c07/pkg/executor/projection.go#L125-L181

Reply:
> it would be better to add the introduction in this PR like
> 
> https://github.com/pingcap/tidb/blob/4ed493dcc1e3d55dd5af46db49d0b6cd34a24c07/pkg/executor/projection.go#L125-L181

okk